diff --git a/dist/cjs/index.js b/dist/cjs/index.js
index a417cefaf5978aa9f32104ee39e04f57e337c62a..223b1f6b0418ba926a764a193023000a4910675b 100644
--- a/dist/cjs/index.js
+++ b/dist/cjs/index.js
@@ -1099,6 +1099,16 @@ function nodeResolve(opts = {}) {
       }
     }
 
+    // TypeScript files may import '.mjs' or '.cjs' to refer to either '.mts' or '.cts'
+    for (const [ext, tsExt] of [
+      ['.mjs', '.mts'],
+      ['.cjs', '.cts']
+    ]) {
+      if (importee.endsWith(ext) && extensions.includes(tsExt)) {
+        importSpecifierList.push(importee.replace(/\.([mc])js$/, '.$1ts'));
+      }
+    }
+
     const warn = (...args) => context.warn(...args);
     const isRequire = custom && custom['node-resolve'] && custom['node-resolve'].isRequire;
     const exportConditions = isRequire ? conditionsCjs : conditionsEsm;
