diff --git a/node_modules/postcss-color-mod-function/index.cjs.js b/node_modules/postcss-color-mod-function/index.cjs.js
index 57e6305..f67ccd4 100644
--- a/node_modules/postcss-color-mod-function/index.cjs.js
+++ b/node_modules/postcss-color-mod-function/index.cjs.js
@@ -1713,7 +1713,7 @@ const alphaMatch = /^a(lpha)?$/i;
 const alphaBlueGreenRedMatch = /^(a(lpha)?|blue|green|red)$/i;
 const blacknessLightnessSaturationWhitenessMatch = /^(b(lackness)?|l(ightness)?|s(aturation)?|w(hiteness)?)$/i;
 const blendMatch = /^blenda?$/i;
-const colorModMatch = /^color-mod$/i;
+const colorModMatch = /^color(?:-mod)?$/i;
 const colorSpaceMatch = /^(hsl|hwb|rgb)$/i;
 const contrastMatch = /^contrast$/i;
 const hexColorMatch$1 = /^#(?:([a-f0-9])([a-f0-9])([a-f0-9])([a-f0-9])?|([a-f0-9]{2})([a-f0-9]{2})([a-f0-9]{2})([a-f0-9]{2})?)$/i;
@@ -1779,7 +1779,7 @@ var index = postcss.plugin('postcss-color-mod-function', opts => {
     }()
   );
 });
-const colorModFunctionMatch = /(^|[^\w-])color-mod\(/i;
+const colorModFunctionMatch = /(^|[^\w-])color(?:-mod)?\(/i;
 
 module.exports = index;
 //# sourceMappingURL=index.cjs.js.map
diff --git a/node_modules/postcss-color-mod-function/index.es.mjs b/node_modules/postcss-color-mod-function/index.es.mjs
index 869795d..c6507fa 100644
--- a/node_modules/postcss-color-mod-function/index.es.mjs
+++ b/node_modules/postcss-color-mod-function/index.es.mjs
@@ -1709,7 +1709,7 @@ const alphaMatch = /^a(lpha)?$/i;
 const alphaBlueGreenRedMatch = /^(a(lpha)?|blue|green|red)$/i;
 const blacknessLightnessSaturationWhitenessMatch = /^(b(lackness)?|l(ightness)?|s(aturation)?|w(hiteness)?)$/i;
 const blendMatch = /^blenda?$/i;
-const colorModMatch = /^color-mod$/i;
+const colorModMatch = /^color(?:-mod)?$/i;
 const colorSpaceMatch = /^(hsl|hwb|rgb)$/i;
 const contrastMatch = /^contrast$/i;
 const hexColorMatch$1 = /^#(?:([a-f0-9])([a-f0-9])([a-f0-9])([a-f0-9])?|([a-f0-9]{2})([a-f0-9]{2})([a-f0-9]{2})([a-f0-9]{2})?)$/i;
@@ -1775,7 +1775,7 @@ var index = postcss.plugin('postcss-color-mod-function', opts => {
     }()
   );
 });
-const colorModFunctionMatch = /(^|[^\w-])color-mod\(/i;
+const colorModFunctionMatch = /(^|[^\w-])color(?:-mod)?\(/i;
 
 export default index;
 //# sourceMappingURL=index.es.mjs.map
